packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -588,11 +591,19 @@ export default class TradeStore extends BaseStore {
                     this.forgetAllProposal();
                     this.purchase_info = response;
                     this.enablePurchase();
+                    this.throttleDisableIsPurchasingContract();
                 })
             );
         }
     }
 
+    throttleDisableIsPurchasingContract = throttle(this.disableIsPurchasingContract, 100);
+
+    @action.bound
+    disableIsPurchasingContract() {
+        this.is_purchasing_contract = false;
```

Comment:
Can we just use `is_purchase_enabled` instead of `is_purchasing_contract`?

Reply:
@zaki-binary well I believe we use that in another way, so properly we can combine them together, but need a great deal of time to check and refactoring to get sure `is_purchase_enabled` treat expected. This way, we have a quick fix for it, to see if it work or not. then if work, we can optimize it and refactor it.